src/DI/Extension.php

```diff
@@ -0,0 +1,40 @@
+<?php declare(strict_types = 1);
+
+namespace Pd\SecurityHeaders\DI;
+
+final class Extension extends \Nette\DI\CompilerExtension
+{
+
+	/**
+	 * @var string
+	 */
+	private $presenterHook;
+
+
+	public function loadConfiguration()
+	{
+		parent::loadConfiguration();
+
+		$config = $this->getConfig();
+
+		if ( ! isset($config['presenterHook'])) {
+			throw new \InvalidArgumentException(\sprintf('Extension "%s" nemá nastavenou hodnotu "presenterHook"', $this->name));
```

Comment:
Nedávala by tady větší smysl InvalidStateException? Tj. buď `\Nette\InvalidStateException` nebo si vytvořit nějakou svoji InvalidStateException, která bude dědit `\RuntimeException` nebo vyhodit přímo `\RuntimeException`.

Reply:
Nebylyo by vhodný přímo v tomto repositáři mít nějakou výchozí implementaci tohoto `presenterHook`? Něco jako je v základu `HeadersSetup`, potom by se to ze základu mohlo smazat. Stejně většinou v tom `presenterHook` potřebujš jenom naházet všechny hlavičky do response. Potom by parametr `presenterHook` mohla být volitelný. V takovém případě bys ale musel vytvořit nějaké rozhraní `IHeadersFactory`, které by vytvářelo hlavičky. Toto rozhraní by už bylo poivné implementovat a uvést v konfiguraci tohoto rozšíření. Co ty na to? Potom jediné, co by se v základu muselo udělat je ta implementace `IHeadersFactory`.